### PR TITLE
feat: ability to delete terminated workstations

### PR DESCRIPTION
**Pull request by sg-s**
## changes

- [x] ability to delete terminated workstations
- [x] ability to delete any workstation
- [x] refactor: single method to make requests with any verb
- [x] expanded testing to py 3.13

### CODE REVIEW

File: src/platform/api.py [hunk 1]
```diff
@@ -35,13 +46,25 @@ def _make_get_request(endpoint: str) -> dict:
...
+    if verb == "GET":
+        response = requests.get(
+            url,
+            headers=headers,
+        )
+    elif verb == "POST":
+        response = requests.post(
+            url,
+            headers=headers,
+        )
```
Comment (kennyjwilli) [inline src/platform/api.py:58]: Perhaps nicer if you use [`requests.request`](https://requests.readthedocs.io/en/latest/api/#requests.request)?

File: src/platform/api.py [hunk 2]
```diff
@@ -51,29 +74,78 @@ def _make_get_request(endpoint: str) -> dict:
         return data
 
 
+@beartype
+def delete_workstation(workstation: str):
+    """delete a workstation"""
+
+    return _make_request(
+        f"computebenches/{_get_org_id()}/{workstation}:delete", verb="POST"
```
Comment (kennyjwilli) [inline src/platform/api.py:82]: Probably not in scope here, but client pattern? 😁

File: src/platform/api.py [hunk 3]
```diff
@@ -51,29 +74,78 @@ def _make_get_request(endpoint: str) -> dict:
...
+        if ws["attributes"]["status"] == "TERMINATED"
+    ]
+
+    if len(names) == 0:
+        print("No workstations to delete")
+        return
+
+    org_id = _get_org_id()
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```
Comment (kennyjwilli) [inline src/platform/api.py:104]: Can you bound the size of the thread pool? Or is it by default?

**Review by kennyjwilli — APPROVED** [review]
General feedback, non-blocking. LGTM.

File: src/platform/api.py (hunk 1 above)
Comment (sg-s) [reply src/platform/api.py:58]: lol much nicer :)

File: src/platform/api.py (hunk 3 above)
Comment (sg-s) [reply src/platform/api.py:104]: the default is ncores

File: src/platform/api.py (hunk 2 above)
Comment (sg-s) [reply src/platform/api.py:82]: yes....also needs to use the autogenerated low level sdk

File: src/platform/api.py (hunk 1 above)
Comment (sg-s) [reply src/platform/api.py]: fixed

**Pull request merged**